File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/TestCaseActionAwaitInvocation.cs
namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class TestCaseActionAwaitInvocation : TestCaseAction
    {
        public int? InvocationIndex { get; set; }

        public object ResponseValue { get; set; } = false;

        public Dictionary<string, string>? Metadata { get; set; }

        public TestCaseCatch? Catch { get; set; }
    }
}

File: go/test/protocol/test_case_action_await_invocation.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

import (
	"gopkg.in/yaml.v3"
)

type testCaseActionAwaitInvocation struct {
	InvocationIndex int                `yaml:"invocation-index"`
	ResponseValue   any                `yaml:"response-value"`
	Metadata        *map[string]string `yaml:"metadata"`
	Catch           *TestCaseCatch     `yaml:"catch"`
}

type TestCaseActionAwaitInvocation struct {
	testCaseActionAwaitInvocation
}

func (actionAwaitInvocation *TestCaseActionAwaitInvocation) UnmarshalYAML(
	node *yaml.Node,
) error {
	*actionAwaitInvocation = TestCaseActionAwaitInvocation{}

	actionAwaitInvocation.ResponseValue = false

	return node.Decode(&actionAwaitInvocation.testCaseActionAwaitInvocation)
}
